pkg/controller/vsphere/actuator_test.go

```diff
@@ -0,0 +1,287 @@
+		name      string
+		error     string
+		operation func(actuator *Actuator, machine *machinev1.Machine)
+		event     string
+	}{
+		{
+			name: "Create machine event failed on invalid machine scope",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Create(nil, machine)
+			},
+			event: "test: failed to create scope for machine: test: machine scope require a context",
+		},
+		{
+			name: "Create machine event failed, reconciler's create failed",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				machine.Labels[machinev1.MachineClusterIDLabel] = ""
+				actuator.Create(context.Background(), machine)
+			},
+			event: "test: reconciler failed to Create machine: test: failed validating machine provider spec: test: missing \"machine.openshift.io/cluster-api-cluster\" label",
+		},
+		{
+			name: "Create machine event succeed",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Create(context.Background(), machine)
+			},
+			event: "Created Machine test",
+		},
+		{
+			name: "Update machine event failed on invalid machine scope",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Update(nil, machine)
+			},
+			event: "test: failed to create scope for machine: test: machine scope require a context",
+		},
+		{
+			name: "Update machine event failed, reconciler's update failed",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				machine.Labels[machinev1.MachineClusterIDLabel] = ""
+				actuator.Update(context.Background(), machine)
+			},
+			event: "test: reconciler failed to Update machine: test: failed validating machine provider spec: test: missing \"machine.openshift.io/cluster-api-cluster\" label",
+		},
+		{
+			name: "Update machine event succeed and only one event is created",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Update(context.Background(), machine)
+				actuator.Update(context.Background(), machine)
+			},
+			event: "Updated Machine test",
+		},
+		{
+			name: "Delete machine event failed on invalid machine scope",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Delete(nil, machine)
+			},
+			event: "test: failed to create scope for machine: test: machine scope require a context",
+		},
+		{
+			name: "Delete machine event failed, reconciler's delete failed",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Delete(context.Background(), machine)
+			},
+			event: "test: reconciler failed to Delete machine: destroying vm in progress, reconciling",
+		},
+		{
+			name: "Delete machine event succeed",
+			operation: func(actuator *Actuator, machine *machinev1.Machine) {
+				actuator.Delete(context.Background(), machine)
+			},
+			event: "Deleted machine test",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
```

Comment:
If you move this to the top of this test function, you could use it in all of the test cases rather than writing `context.Background()` out for each one, or you could change the function signature to include a context and pipe it through that way

```
func(ctx context.Context, actuator *Actuator, machine *machinev1.Machine) {
  actuator.Delete(ctx, machine)
}
```